assets/section-main-product.css

```diff
@@ -496,7 +496,7 @@ a.product__text {
   }
 
   .product__media-list .product__media-item {
-    width: calc(100% - 3rem);
+    width: calc(100% - 3rem - var(--grid-mobile-horizontal-spacing));
   }
 }
```

Comment:
I think there is an additional change we can add. On a single media product, we get the controls that show up even though there is only 1 image. 

It happens because of the way we set the `media_count` value: 
```liquid
assign media_count = product.media.size
if section.settings.hide_variants
  assign media_count = media_count | minus: variant_images.size | plus: 1
endif
```

Which seem wrong. As it means that in our single image product case, we're doing: `1 - 0 + 1 = 2` when it should just be one. 
The other situation could be that we have a product with a single variant and the media is attached to it. Then it's fine cause it ends up being `1 - 1 + 1`. 

So I think we could add this to the if statement: `if section.settings.hide_variants and product.media.size > 1` 

[here is a video of this](https://screenshot.click/13-48-9y1g9-721m7.mp4) ^

Reply:
And yes, the padding issue with the first variant is still an issue here. I added a note in the PR for reference.